### PR TITLE
Update QA Document

### LINKED ISSUE #351: QA plan does not distinguish between UTs and E2E tests
- [x] Add End-to-end Testing and Integration Testing columns. 
- [x] Move some entries under Unit Testing to End-to-end Testing.
- [x] Update with new info re. UTs

### PR DESCRIPTION
**Pull request by smarshall-spitzbart**
Closes #351

This PR is intended to compliment #342. I split these diffs out so that extra attention could be put into verifying the  updates to `quality_assurance.md` are correct.

### CODE REVIEW

**Review by sainoe — APPROVED** [review]
LGTM

File: docs/quality_assurance.md
Comment (danwt) [inline docs/quality_assurance.md:47]: Ultra nit: I'd be wary of specifying test names as it might change in future.

File: docs/quality_assurance.md
Comment (danwt) [inline docs/quality_assurance.md:65]: Ditto: I'm a bit wary of linking/specifying test names. (Although I see this was already done before your PR so not on you!)

**Review by danwt — APPROVED** [review]
Thanks for the updates. I'll approve but think we should wait for Marius.

File: docs/quality_assurance.md
Comment (smarshall-spitzbart) [reply docs/quality_assurance.md:65]: Hmm yea I agree with you that linking hardcoded test names and specific lines of code is not ideal. On the other hand, the links may save people time searching through files, and I can't really think of a better alternative. Lmk if one exists 🤔

**Comment by smarshall-spitzbart** [comment]
@mpoke I'll wait on your review before merging

**Review by jtremback — APPROVED** [review]
LGTM

**Pull request merged**
